### ISSUE TITLE
Multiple Filesystems: When creating second filesystem need actively restart csi-cephfsplugin-provisioner 

### ISSUE THREAD

**Issue by glyslxq** [rook/rook#10623]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**

**Expected behavior:**

**How to reproduce it (minimal and precise):**

**File(s) to submit**:

* Cluster CR (custom resource), typically called `cluster.yaml`, if necessary
* Operator's logs, if necessary
* Crashing pod(s) logs, if necessary

csi-cephfsplugin-provisioner logs
```
E0724 02:05:42.557156       1 controller.go:957] error syncing claim "7cfd671b-68bd-477b-a99f-1b9ea04533af": failed to provision volume with StorageClass "rook-cephfs-phptest": rpc error: code = Internal desc = rados: ret=-2, No such file or directory: "subvolume group 'csi' does not exist"
```

After actively restarting CSI cephfsplugin provider, it works normally
```
I0724 02:06:45.723164       1 event.go:285] Event(v1.ObjectReference{Kind:"PersistentVolumeClaim", Namespace:"dev", Name:"phptest-pvc", UID:"7cfd671b-68bd-477b-a99f-1b9ea04533af", APIVersion:"v1", ResourceVersion:"459383221", FieldPath:""}): type: 'Warning' reason: 'ProvisioningFailed' failed to provision volume with StorageClass "rook-cephfs-phptest": rpc error: code = Internal desc = rados: ret=-2, No such file or directory: "subvolume group 'csi' does not exist"
I0724 02:07:04.946520       1 controller.go:1337] provision "dev/phptest-pvc" class "rook-cephfs-phptest": started
I0724 02:07:04.946703       1 event.go:285] Event(v1.ObjectReference{Kind:"PersistentVolumeClaim", Namespace:"dev", Name:"phptest-pvc", UID:"7cfd671b-68bd-477b-a99f-1b9ea04533af", APIVersion:"v1", ResourceVersion:"459383221", FieldPath:""}): type: 'Normal' reason: 'Provisioning' External provisioner is provisioning volume for claim "dev/phptest-pvc"
W0724 02:07:04.983642       1 controller.go:934] Retrying syncing claim "7cfd671b-68bd-477b-a99f-1b9ea04533af", failure 8
E0724 02:07:04.983684       1 controller.go:957] error syncing claim "7cfd671b-68bd-477b-a99f-1b9ea04533af": failed to provision volume with StorageClass "rook-cephfs-phptest": rpc error: code = Internal desc = rados: ret=-2, No such file or directory: "subvolume group 'csi' does not exist"
I0724 02:07:04.984234       1 event.go:285] Event(v1.ObjectReference{Kind:"PersistentVolumeClaim", Namespace:"dev", Name:"phptest-pvc", UID:"7cfd671b-68bd-477b-a99f-1b9ea04533af", APIVersion:"v1", ResourceVersion:"459383221", FieldPath:""}): type: 'Warning' reason: 'ProvisioningFailed' failed to provision volume with StorageClass "rook-cephfs-phptest": rpc error: code = Internal desc = rados: ret=-2, No such file or directory: "subvolume group 'csi' does not exist"
I0724 02:07:49.723568       1 controller.go:1337] provision "dev/phptest-pvc" class "rook-cephfs-phptest": started
I0724 02:07:49.723684       1 event.go:285] Event(v1.ObjectReference{Kind:"PersistentVolumeClaim", Namespace:"dev", Name:"phptest-pvc", UID:"7cfd671b-68bd-477b-a99f-1b9ea04533af", APIVersion:"v1", ResourceVersion:"459383221", FieldPath:""}): type: 'Normal' reason: 'Provisioning' External provisioner is provisioning volume for claim "dev/phptest-pvc"
W0724 02:07:49.754520       1 controller.go:934] Retrying syncing claim "7cfd671b-68bd-477b-a99f-1b9ea04533af", failure 9
E0724 02:07:49.754552       1 controller.go:957] error syncing claim "7cfd671b-68bd-477b-a99f-1b9ea04533af": failed to provision volume with StorageClass "rook-cephfs-phptest": rpc error: code = Internal desc = rados: ret=-2, No such file or directory: "subvolume group 'csi' does not exist"
I0724 02:07:49.754618       1 event.go:285] Event(v1.ObjectReference{Kind:"PersistentVolumeClaim", Namespace:"dev", Name:"phptest-pvc", UID:"7cfd671b-68bd-477b-a99f-1b9ea04533af", APIVersion:"v1", ResourceVersion:"459383221", FieldPath:""}): type: 'Warning' reason: 'ProvisioningFailed' failed to provision volume with StorageClass "rook-cephfs-phptest": rpc error: code = Internal desc = rados: ret=-2, No such file or directory: "subvolume group 'csi' does not exist"
I0724 02:09:22.977793       1 csi-provisioner.go:139] Version: v3.1.0
I0724 02:09:22.977838       1 csi-provisioner.go:162] Building kube configs for running in cluster...
I0724 02:09:23.921372       1 csi-provisioner.go:139] Version: v3.1.0
I0724 02:09:23.921450       1 csi-provisioner.go:162] Building kube configs for running in cluster...
I0724 02:09:24.923966       1 common.go:111] Probing CSI driver for readiness
I0724 02:09:24.928532       1 csi-provisioner.go:278] CSI driver does not support PUBLISH_UNPUBLISH_VOLUME, not watching VolumeAttachments
I0724 02:09:24.930755       1 leaderelection.go:248] attempting to acquire leader lease rook-ceph/rook-ceph-cephfs-csi-ceph-com...
I0724 02:09:25.013981       1 common.go:111] Probing CSI driver for readiness
I0724 02:09:25.018517       1 csi-provisioner.go:278] CSI driver does not support PUBLISH_UNPUBLISH_VOLUME, not watching VolumeAttachments
I0724 02:09:25.019743       1 leaderelection.go:248] attempting to acquire leader lease rook-ceph/rook-ceph-cephfs-csi-ceph-com...
I0724 02:09:44.686315       1 leaderelection.go:258] successfully acquired lease rook-ceph/rook-ceph-cephfs-csi-ceph-com
I0724 02:09:44.787733       1 controller.go:811] Starting provisioner controller rook-ceph.cephfs.csi.ceph.com_csi-cephfsplugin-provisioner-5b779dfcf8-kz5vk_b68deef6-61dc-4fed-be25-71bb2f47d61d!
I0724 02:09:44.787821       1 clone_controller.go:66] Starting CloningProtection controller
I0724 02:09:44.787828       1 volume_store.go:97] Starting save volume queue
I0724 02:09:44.788097       1 clone_controller.go:82] Started CloningProtection controller
I0724 02:09:44.888899       1 controller.go:1337] provision "dev/phptest-pvc" class "rook-cephfs-phptest": started
I0724 02:09:44.889822       1 controller.go:860] Started provisioner controller rook-ceph.cephfs.csi.ceph.com_csi-cephfsplugin-provisioner-5b779dfcf8-kz5vk_b68deef6-61dc-4fed-be25-71bb2f47d61d!
I0724 02:09:44.890903       1 event.go:285] Event(v1.ObjectReference{Kind:"PersistentVolumeClaim", Namespace:"dev", Name:"phptest-pvc", UID:"7cfd671b-68bd-477b-a99f-1b9ea04533af", APIVersion:"v1", ResourceVersion:"459383221", FieldPath:""}): type: 'Normal' reason: 'Provisioning' External provisioner is provisioning volume for claim "dev/phptest-pvc"
I0724 02:09:45.052416       1 controller.go:1442] provision "dev/phptest-pvc" class "rook-cephfs-phptest": volume "pvc-7cfd671b-68bd-477b-a99f-1b9ea04533af" provisioned
I0724 02:09:45.052456       1 controller.go:1455] provision "dev/phptest-pvc" class "rook-cephfs-phptest": succeeded
I0724 02:09:45.076545       1 event.go:285] Event(v1.ObjectReference{Kind:"PersistentVolumeClaim", Namespace:"dev", Name:"phptest-pvc", UID:"7cfd671b-68bd-477b-a99f-1b9ea04533af", APIVersion:"v1", ResourceVersion:"459383221", FieldPath:""}): type: 'Normal' reason: 'ProvisioningSucceeded' Successfully provisioned volume pvc-7cfd671b-68bd-477b-a99f-1b9ea04533af
```

**Environment**:
* OS (e.g. from /etc/os-release): Ubuntu 20.04.4 LTS
* Kernel (e.g. `uname -a`): Linux 5.4.0-109-generic #123-Ubuntu SMP Fri Apr 8 09:10:54 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
* Cloud provider or hardware configuration:
* Rook version (use `rook version` inside of a Rook Pod): 1.9.7
* Storage backend version (e.g. for ceph do `ceph -v`): ceph version 16.2.9 (4c3647a322c0ff5a1dd2344e039859dcbd28c830) pacific (stable)
* Kubernetes version (use `kubectl version`): 1.20.15
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift): Community
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/latest/Troubleshooting/ceph-toolbox/#interactive-toolbox)): HEALTH_OK

**Comment by Madhu-1** [comment]
@glyslxq have you deleted and recreated the filesystem with the same name? If yes this is expected if no please please let me know below is the case to hit this one

* Create a filesystem with the name `myfs-a` (example) 
* Create Storageclass and PVC from filesystem `myfs-a`
* Create filesystem with name `myfs-b` (example)
* Create StorageClass and PVC from `myfs-b` -> the PVC creation is failing here

**Comment by glyslxq** [comment]
@Madhu-1 yes,I deleted the file system with the same name and recreated it.

**Comment by Madhu-1** [comment]
Yes, if you delete and recreate the filesystem with the same name it's expected. You need to restart the csi-provisioner pods.

**Comment by t3hmrman** [comment]
Just commenting to confirm @Madhu-1's comment, I needed to restart the CSI provisioner pod after creating a CephFS filesystem & storageclass:

```console
kubectl rollout restart deploy/csi-cephfsplugin-provisioner -n rook-ceph
```

**Comment by Madhu-1** [comment]
> Just commenting to confirm @Madhu-1's comment, I needed to restart the CSI provisioner pod after creating a CephFS filesystem & storageclass:
> 
> ```
> kubectl rollout restart deploy/csi-cephfsplugin-provisioner -n rook-ceph
> ```

@t3hmrman Yes, let me see if can make fix this one and avoid restarting the csi provisioner.

**Comment by Madhu-1** [comment]
https://github.com/ceph/ceph-csi/pull/3375 make thins better.

**Comment by Madhu-1** [comment]
This will be available in 3.7.2 release (next week)
